### ISSUE TITLE
Subflow parent nodes doesn't update size when nodes are re-initialized

### ISSUE THREAD

**Issue by moklick** [xyflow/xyflow#3886]
If you specify a size for a parent node it should always be overwritten when one of the child nodes uses the `expandParent` option. As you can see in the example, the size gets overwritten initially but not when you reset the nodes (with the same set of nodes).

### Example 

https://codesandbox.io/p/sandbox/subflow-bug-dlqsnr?file=/src/Flow.tsx

### Steps to Reproduce the Bug or Issue

1. open codesandbox link
2. click "reset nodes"
3. parent node has a different size

### Expected behavior

parent node should keep the adjusted size.

### Platform

- OS: macOs 
- Browser: Chrome
- Version: 11.10.3

### Additional context

Initially we measure all nodes and send the updates to the `applyNodeChanges` function. There we adjust the size of the parent node and everything looks good. If you then use the same nodes to re-initialize the flow, the child node is already there and we won't measure it again. That's why there is no change that could lead to a resizing of the parent node.

**Comment by mcupito** [comment]
Is there any update on this? Thanks!

**Comment by moklick** [comment]
We are working on a fix. For now the only workaround on user side is to reset the nodes before you set new ones:

```js
// reset with an empty array
setNodes([]);

setTimeout(()=>{
  // set actual nodes
  setNodes(nodes)
},0)
```

**Comment by moklick** [comment]
Fix is released in 11.10.4.
